python/lsst/ts/salobj/topics/read_topic.py

```diff
@@ -434,7 +435,10 @@ async def close(self):
         self._data_queue.clear()
 
     async def aget(self, timeout=None):
-        """Get the current message, if any, else wait for the next message.
+        """Get the most recent message, or wait for data if none available
+        (`has_data` False).
+
+        This method neither reads from, nor flushes, the queue.
```

Comment:
Then it comes from the cache?

Reply:
It comes from the `_current_data` attribute, which is one source of cached data, the queue being the other.